alpakaConfig.cmake

```diff
@@ -238,6 +236,17 @@ ELSE()
     ENDIF()
 ENDIF()
 
+#-------------------------------------------------------------------------------
+# Find Threads.
+IF(NOT MSVC)
```

Comment:
Are we sure that this is required on Mac as well?

Reply:
before we added pthreads for MAC to the linker: https://github.com/ComputationalRadiationPhysics/alpaka/pull/866/files#diff-cd45f124e237a4bd8df18ffab83a5664L224
So this PR will only additionally add it to the compiler too.